### ISSUE TITLE
Research on LG Smart Watch integration for heart rate and steps measurements

### ISSUE THREAD

**Issue by cretualexandru** [cami-project/cami-project#95]
## Why
We need to research in order to find all the ways we can properly read the heart rate and steps measurements from the LG W150 Smart Watch.

After the research is done, we need to choose the best way to do this, a trade off between ease of implementation, accuracy of measurements and usability.

## What
* [x] Find all the possible implementations
* [x] Document them including advantages and disadvantages
* [x] Choose the best implementation

**Comment by cretualexandru** [comment]
We have two possibilities:

#### 1. Create our own Android application to read the measurements from the sensors and use our API to insert them in the database.

Advantages:
- we can control the implementation to get as accurate measurements as possible
- we can create a simple and easy to use application, implementing only the needed functionalities
- being our own implementation, we can assure that we create a properly functioning application and we know it's complete behavior

Disadvantages:
- more time needed to learn to create Android and Android Wear applications
- more time needed to actually create the application

#### 2. Use a 3rd party Android application to check the heart rate on a predefined time interval. This application will send the measurements to Google Fit database, and we would have to read them from Google Fit using their REST API.

Advantages:
- we save time because we don't have to create a new Android application

Disadvantages:
- 3rd party applications are created for other purposes, having a lot of useless functionalities for our use case, thus adding an overhead
- 3rd party applications may be unreliable, they can have an unexpected behavior, that may be problematic for old people
- 3rd party applications may need special configuration, which would damage the usability
- some 3rd party applications give very non accurate measurements
- very few 3rd party applications that meet 4 important requirements:
 - provide heart check measurements on a time interval
 - provide accurate measurements
 - be (relatively) easy to use
 - work properly

The best 3rd application I have found is Cinch.

Personally, I would go with the first possibility, though this would mean to step into the unknown. I'm looking forward to hearing your opinion too, @aismail @paraschivionut @rtud .

**Comment by cretualexandru** [comment]
We chose to use the second way, alongside the [Cinch](https://play.google.com/store/apps/details?id=com.ryansteckler.perfectcinch&hl=en) application which seems to be the best choice for now. Cinch will send the collected data to Google Fit, and we'll use the REST API to get the data from Google Fit and store it in the CAMI Cloud.

We found out, though, that using this implementation will not give us a real-time heart-rate and steps monitoring, as it depends of "Google Fit Data" Auto Sync. It seems that an auto sync is made once every two hours.

Closing!
